### ISSUE TITLE
Remote Debug Breakpoint Error (Docker)

### ISSUE THREAD

**Issue by DonJayamanne** [Microsoft/ptvsd#1155]
@sgtoj commented on [Mon Feb 11 2019](https://github.com/Microsoft/vscode-python/issues/4349)

## Environment data

- VS Code version: 1.32.0-insiders
- Extension version (available under the Extensions sidebar): 2019.1.0
- OS and version: Windows 1803
- Python: 3.7.2 64-bit (user install)
- Python Packages: ptvsd 4.2.3
- Docker:   Docker (for Windows) version 18.09.1, build 4c52b90
- Docker Compose:  docker-compose version 1.23.2, build 1110ad01

## Expected behaviour

Attach to debugger server with breakpoints.

## Actual behaviour

Error is thrown that it cannot add breakpoint to non-existing file. It is mapping to the wrong path. 
```
Recreating python_app ... done
Attaching to python_app
python_app | pydev debugger: warning: trying to add breakpoint to file that does not exist: /opt/app/c:/Users/mydevuser/Downloads/py_debug_example/app/start.py (will have no effect)
python_app | hello world!
python_app | Traceback (most recent call last):
python_app |   File "/usr/local/lib/python3.7/site-packages/ptvsd/wrapper.py", line 1422, in done
python_app |     fut.result()
python_app |   File "/usr/local/lib/python3.7/site-packages/ptvsd/futures.py", line 40, in result
python_app |     reraise(self._exc_info)
python_app |   File "/usr/local/lib/python3.7/site-packages/ptvsd/reraise3.py", line 8, in reraise
python_app |     raise exc_info[1].with_traceback(exc_info[2])
python_app |   File "/usr/local/lib/python3.7/site-packages/ptvsd/futures.py", line 157, in callback
python_app |     x = next(it)
python_app |   File "/usr/local/lib/python3.7/site-packages/ptvsd/wrapper.py", line 1677, in on_threads
python_app |     _, _, resp_args = yield self.pydevd_request(cmd, '')
python_app |   File "/usr/local/lib/python3.7/site-packages/ptvsd/wrapper.py", line 1451, in pydevd_request
python_app |     return self._pydevd_request(self.loop, cmd_id, args, is_json=is_json)
python_app |   File "/usr/local/lib/python3.7/site-packages/ptvsd/wrapper.py", line 514, in pydevd_request
python_app |     raise EOFError
python_app | EOFError
python_app exited with code 0
```

## Steps to reproduce:

Clone [py_debug_example](https://github.com/sgtoj/py_debug_example) and follow its instruction.

## Logs

```
##########Linting Output - pylint##########

--------------------------------------------------------------------

Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)



[Error - 9:11:29 AM] One or more errors occurred. (failed to parse file file:///c:/Users/mydevuser/Downloads/py_debug_example/start.py)
```   

## Traceback w/ Source Map Enable

See raw [traceback with source map enable](https://pastebin.com/raw/c8FACvdu) output hosted by pastebin.

---

@sgtoj commented on [Mon Feb 11 2019](https://github.com/Microsoft/vscode-python/issues/4349#issuecomment-462430294)

I have review several GH related issues across this project and the `ptvsd` project. Those other issues have been closed (as of late last week).  Hence the reason for this new issue.

_Please let me know if there is anything else I can do or provide to assist with this issue._

## Troubleshooting 

- reviewed several github issues
- tried setting `localRoot` to all lower case path, exact case path, and `${workspaceFolder}`
- issue affects 3 different windows 10 systems (1709, 1803, and latest Insider _fast_ build)

## Reviewed Issues

- #3076 - not using django
- #3567 - not using cygwin; explicitly setting `pathMappings`
- #2991 - using `python -m ptvsd` method
- Microsoft/ptvsd#1059 - similar but quite the same issue; not using anaconda
- #3568 - using `attach` method
- Microsoft/vscode#67523 - tried all lower case path and exact case path as well as `${workspaceFolder}`
- Microsoft/ptvsd#1115 - not importing `ptvsd` in script (calling it launch time); not using django

## VS Code Settings

These are python specific _user settings_ for VS Code. The _workspace settings_ are committed with the example git project.

```
  "python.formatting.provider": "yapf",
  "python.jediEnabled": false,
  "python.linting.pylintEnabled": true,
  "python.workspaceSymbols.ctagsPath": "C:\\ctags\\ctags.exe",
```

---

@webartoli commented on [Tue Feb 19 2019](https://github.com/Microsoft/vscode-python/issues/4349#issuecomment-465187796)

Same issue on different environment:

* VS Code version: 1.31.1
* Extension version (available under the Extensions sidebar): 2019.1.0
* OS and version: Windows 1809
* Python: 3.6.8
* Python Packages: ptvsd 4.2.3
* Docker:   Docker (for Windows) version 18.09.2, build 6247962
* Docker Compose:  docker-compose version 1.24.0-rc1, build 0f3d4dda

Application is running on container with the following docker-compose

```yaml
service-name:
  build:
    context: .
    dockerfile: "Dockerfile"
  ports:
    - "5678:5678"
  command: python -m ptvsd --host 0.0.0.0 --port 5678 service_bootstrap.py
```

**Comment by omartin2010** [comment]
Also, not sure if entierly related, but when I'm putting the right syntax (small caps c:\\Users\\...\\workind_dir, then I connect, and don't see the error messages on the path, however I see a bunch of threads that are running, and one is said to be Paused, yet the editor that normally displays the breakpoint doesn't show up.  See here : ![image](https://user-images.githubusercontent.com/11371870/53048511-e8e0d980-3462-11e9-8805-a1aa432c528e.png)

I see the one that shows the original, local file, but not the remote identical copy of the file that I normally see in a newly displayed code window (that if I hover the mouse over, I would see it's on, for example /home/user/...). And if I do right click on the MainThread I have the options to step over, into, ... and the debug displays seem to suggest that it's working. However it's not displayed and not being displayed makes it hard to follow. This was working before, and I'm not sure under what specific conditions this happens but I have a constant repro now (and am internal to the team). Just now, after about 5 minutes of not touching the VS Code window as I was typing  here, it finally showed, but it is ultra slow whenever I press F10, and I don't get control back.

Moreover, if I stop the debugging sessions, the Call Stack keeps showing as busy and the Locals variables seem like they're also on hold somehow. See here : 
![image](https://user-images.githubusercontent.com/11371870/53048793-948a2980-3463-11e9-8f6a-5c393f031be5.png)

**Comment by karthiknadig** [comment]
@sgtoj I had to change ./app/app.py to ./app/start.py . Alos, we recently fixed an issue where we had a race with path mapping and setting breakpoints. With ptvsd=4.2.4 and the rename from app.py to start.py it does not repro. 
![image](https://user-images.githubusercontent.com/3840081/53050438-28e99180-344e-11e9-8c8d-e86dbe67289f.png)

**Comment by karthiknadig** [comment]
@omartin2010 That looks like a different issue. Can you file a separate bug with details about your project and environment.

**Comment by omartin2010** [comment]
Opened a new one, thanks @karthiknadig !

**Comment by karthiknadig** [comment]
@sgtoj Please reopen this issue if this issue continues to occur with ptvsd 4.2.4+
